### PR TITLE
Update Example: react/virtualized-rows

### PR DESCRIPTION
**Pull request by dgildeh**
I did this update out of frustration trying to find examples of doing this with the V3 version of react-virtual and also figuring out how to virtualize the table using the Window scrollbar (useWindowVirtualizer).

The example has been updated to switch between two versions of the virtualized table, one a fixed height container as in the original example, and the other using the Window Scrollbar so users can scroll down the page:

* Updated to use latest version of @tanstack/react-virtual: 3.0.0-beta.33
* Migrated existing example to fixedHeightTable.tsx
* Added windowHeightTable.tsx with example of how to render page length virtualized table using useWindowVirtualizer
* A few other fixes that were causing TypeScript errors in my IDE

### CODE REVIEW

File: examples/react/virtualized-rows/src/fixedHeightTable.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  const virtualRows = rowVirtualizer.getVirtualItems()
+  const totalSize = rowVirtualizer.getTotalSize()
+
+  // This is where the magic happens, essentially create a large row with height of total rows
+  // before and after the first/last displayed row so the scroll bar works correctly
+  const paddingTop = virtualRows.length > 0 ? virtualRows?.[0]?.start || 0 : 0
+  const paddingBottom =
+    virtualRows.length > 0
+      ? totalSize - (virtualRows?.[virtualRows.length - 1]?.end || 0)
+      : 0
```
Comment (piecyk) [inline examples/react/virtualized-rows/src/fixedHeightTable.tsx:66]: We can keep those values together

```suggestion
  const [paddingTop, paddingBottom] =
    virtualRows.length > 0
      ? [
          Math.max(0, virtualRows[0].start),
          Math.max(0, rowVirtualizer.getTotalSize() - virtualRows[virtualRows.length - 1].end),
        ]
      : [0, 0]
```

File: examples/react/virtualized-rows/src/fixedHeightTable.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        <tbody>
+          {paddingTop > 0 && (
+            <tr>
+              <td style={{ height: `${paddingTop}px` }} />
+            </tr>
+          )}
+          {virtualRows.map(virtualRow => {
+            const row = rows[virtualRow.index] as Row<Person>
+            return (
+              <tr key={row.id}>
```
Comment (piecyk) [inline examples/react/virtualized-rows/src/fixedHeightTable.tsx:122]:
```suggestion
              <tr key={virtualRow.key}>
```

File: examples/react/virtualized-rows/src/fixedHeightTable.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  })
+
+  const tableContainerRef = React.useRef<HTMLDivElement>(null)
+
+  const { rows } = table.getRowModel()
+  const rowVirtualizer = useVirtualizer({
+    count: rows.length,
+    getScrollElement: () => tableContainerRef.current,
+    estimateSize: () => 54,
+    overscan: 10,
```
Comment (piecyk) [inline examples/react/virtualized-rows/src/fixedHeightTable.tsx:54]: Let's we need to pass correct keys to virtualizer it's important when rows change position
```suggestion
    overscan: 10,
    getItemKey: React.useCallback(index => rows[index].id, [rows])
```

File: examples/react/virtualized-rows/src/fixedHeightTable.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  })
+
+  const virtualRows = rowVirtualizer.getVirtualItems()
+  const totalSize = rowVirtualizer.getTotalSize()
```
Comment (piecyk) [inline examples/react/virtualized-rows/src/fixedHeightTable.tsx:58]: Moved to inline calculation of paddingBottom

File: examples/react/virtualized-rows/src/fixedHeightTable.tsx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+  SortingState,
+  useReactTable,
+} from '@tanstack/react-table'
+import { useVirtualizer } from '@tanstack/react-virtual'
+
+import { Person } from './makeData'
+
+type FixedHeightTableProps = {
+  // The data to render
+  data: any
```
Comment (piecyk) [inline examples/react/virtualized-rows/src/fixedHeightTable.tsx:17]: Why any?

File: examples/react/virtualized-rows/src/windowHeightTable.tsx [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+  React.useLayoutEffect(() => {
+    tableOffsetRef.current = tableContainerRef.current?.offsetTop ?? 0
+  }, [])
+
+  const { rows } = table.getRowModel()
+  const rowVirtualizer = useWindowVirtualizer({
+    count: rows.length,
+    scrollMargin: tableOffsetRef.current,
+    estimateSize: () => 54,
+    overscan: 10,
```
Comment (piecyk) [inline examples/react/virtualized-rows/src/windowHeightTable.tsx:54]: Add getItemKey

File: examples/react/virtualized-rows/src/windowHeightTable.tsx [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+      : 0
+
+  return (
+    <div
+      ref={tableContainerRef}
+      className="container"
+      style={{
+        // Need to add 52px to totalSize to account for header height + container border top and bottom
+        height: rowVirtualizer.getTotalSize() + 52,
+      }}
```
Comment (piecyk) [inline examples/react/virtualized-rows/src/windowHeightTable.tsx:75]: This should not be needed, if header is sticky then it size should be added to scrollMargin

File: examples/react/virtualized-rows/src/windowHeightTable.tsx [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+          {paddingTop > 0 && (
+            <tr>
+              <td style={{ height: `${paddingTop}px` }} />
+            </tr>
+          )}
+          {virtualRows.map(virtualRow => {
+            const row = rows[virtualRow.index] as Row<Person>
+            return (
+              <tr
+                key={row.id}
```
Comment (piecyk) [inline examples/react/virtualized-rows/src/windowHeightTable.tsx:123]:
```suggestion
                key={virtualRow.key}
```

File: examples/react/virtualized-rows/src/windowHeightTable.tsx [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+  const virtualRows = rowVirtualizer.getVirtualItems()
+
+  // We need to add a padding row at top that grows as we scroll down to ensure
+  // the table scrolls down with the page
+  const paddingTop =
+    virtualRows.length > 0
+      ? virtualRows?.[0]?.start
+        ? virtualRows?.[0]?.start - tableOffsetRef.current
+        : 0 || 0
+      : 0
```
Comment (piecyk) [inline examples/react/virtualized-rows/src/windowHeightTable.tsx:66]: We can use almost same logic as in, just here we need to remove also scrollMargin as we use it 

```suggestion
 const [paddingTop, paddingBottom] =
    virtualRows.length > 0
      ? [
          Math.max(0, virtualRows[0].start - rowVirtualizer.options.scrollMargin),
          Math.max(0, rowVirtualizer.getTotalSize() - virtualRows[virtualRows.length - 1].end),
        ]
      : [0, 0]
```

**Review by piecyk — CHANGES_REQUESTED** [review]
Overall looks good, added few comments, please update also file names as those exporting components via convention should be pascal case.

File: examples/react/virtualized-rows/src/fixedHeightTable.tsx (hunk 1 above)
Comment (dgildeh) [reply examples/react/virtualized-rows/src/fixedHeightTable.tsx:66]: Updated but had to add some extra logic to remove undefined errors in TS

File: examples/react/virtualized-rows/src/fixedHeightTable.tsx (hunk 2 above)
Comment (dgildeh) [reply examples/react/virtualized-rows/src/fixedHeightTable.tsx:122]: Done

File: examples/react/virtualized-rows/src/fixedHeightTable.tsx (hunk 3 above)
Comment (dgildeh) [reply examples/react/virtualized-rows/src/fixedHeightTable.tsx:54]: Added, but had to update callback to get rid of type errors

File: examples/react/virtualized-rows/src/fixedHeightTable.tsx (hunk 4 above)
Comment (dgildeh) [reply examples/react/virtualized-rows/src/fixedHeightTable.tsx:58]: I removed totalSize as your combined paddingTop/paddingBottom calculation doesn't use it

File: examples/react/virtualized-rows/src/windowHeightTable.tsx (hunk 8 above)
Comment (dgildeh) [reply examples/react/virtualized-rows/src/windowHeightTable.tsx:123]: Fixed

File: examples/react/virtualized-rows/src/windowHeightTable.tsx (hunk 7 above)
Comment (dgildeh) [reply examples/react/virtualized-rows/src/windowHeightTable.tsx:75]: I removed it - in my original version I wasn't using paddingBottom, now that that's added the container scales to the correct height as needed.

File: examples/react/virtualized-rows/src/windowHeightTable.tsx (hunk 6 above)
Comment (dgildeh) [reply examples/react/virtualized-rows/src/windowHeightTable.tsx:54]: Added

**Comment by dgildeh** [comment]
Fixed all comments and updated filenames to PascalCase.

**Comment by KevinVandy** [comment]
It would be cool to still get fixed height and window examples into the docs. Closing for now because of merge conflicts. Can work with you if you are interested, albeit, a year later.
I started a whole [virtualization guide](https://tanstack.com/table/v8/docs/guide/virtualization) in the table docs that can definitely use these concepts

**Pull request closed without merging**